service/user_http_test.go

```diff
@@ -82,3 +46,56 @@ func makeHTTPCall(method, path, body string, handlerFunc http.HandlerFunc) (reco
 	router.ServeHTTP(recorder, req)
 	return
 }
+
+func (suite *UsersHandlerTestSuite) TestRegisterUserSuccess() {
+	suite.dbMock.On("CreateNewUser", mock.Anything, mock.Anything).Return(db.User{}, nil)
```

Comment:
avoid using `mock.Anything` instead use actual User object
for context you can use `mock.Anything`

Reply:
If we don't use mock.Anything for user then we have to manually fill the user object